lib/tasks/create_printers.rake

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+namespace :printers do
+  desc 'generate list of printers'
+  task create: :environment do |_t|
+    existing_printers = `lpstat -a`
+    Printer.all.each do |printer|
+      if existing_printers.include? printer.name
+      else
+        `sudo lpadmin -p #{printer.name} -v socket://#{printer.name}.internal.sanger.ac.uk -E`
+        puts "Added printer ~> #{printer.name}"
+      end
```

Comment:
This is an if ... else...  structure, but it doesnt have anything in the if... block, could you rewrite it? Or substitute with unless if there is nothing in the if....

Reply:
Is existing_printers a list? or a string?
What would happen if printer.name is a string like '3'. Will it match any printer name with a '3' in its name?

Reply:
This code is in printers_controller already, so we have it here duplicated. Previous changes would need to be applied in both places.
One way to solve it could be to move this functionality into the Printer class, so it can execute after you create a new printer.
Have a look at this doc with callbacks (after_commit, after_save, etc), about options about how you could do it: https://guides.rubyonrails.org/active_record_callbacks.html

Reply:
These issues could be solved by calling the same block defined in app/models/printer. Maybe you can refactor this part to reuse it?